src/com/projectkorra/projectkorra/ability/ElementalAbility.java

```diff
@@ -205,11 +222,23 @@ public static boolean isUndead(final Entity entity) {
 	}
 
 	public static boolean isWater(final Block block) {
-		return block != null ? isWater(block.getType()) : null;
+		if (block == null) {
+			return false;
+		} else if (isWater(block.getType())) {
+			return true;
+		} else if (isWater(block.getBlockData())) {
+			return true;
+		} else {
+			return false;
+		}
+	}
+	
+	public static boolean isWater(final BlockData data) {
+		return (data instanceof Waterlogged) ? ((Waterlogged) data).isWaterlogged() : false;
```

Comment:
In my opinion, we should have the type check here as well. So maybe change `false` to `isWater(data.getType())` or something